client/ayon_maya/plugins/load/load_reference.py

```diff
@@ -239,12 +248,68 @@ def update(self, container, context):
         members = get_container_members(container)
         self._lock_camera_transforms(members)
 
+    def remove(self, container):
+        representation_id: str = container["representation"]
+        project_name: str = container.get(
+            "project_name", get_current_project_name()
+        )
+        product_type = None
+        if representation_id:
+            context: dict = get_representation_context(
+                project_name, representation_id
+            )
+            product_type: str = context["product"]["productType"]
+
+        if product_type == "rig":
+            # No special handling needed for non-rig containers
+            self._remove_rig(container)
+            return
+
+        super().remove(container)
+
+    def _remove_rig(self, container):
```

Comment:
Please add a docstring of what special behavior is needed for the rig products, so it's clear what this method is trying to do.